.spec/4414-parser-layering/context.md

```
@@ -0,0 +1,165 @@
+
+## Root Cause Analysis
+
+### Why the re-exports exist
+
+The re-exports were added to provide a single import path. Instead of:
+```rust
+use perl_lsp_code_actions::CodeActionsProvider;
+use perl_lsp_completion::CompletionProvider;
+use perl_lsp_diagnostics::DiagnosticProvider;
+```
+
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)